src/log/log.c

```diff
@@ -2227,30 +2226,21 @@ __log_write_internal(WT_SESSION_IMPL *session, WT_ITEM *record, WT_LSN *lsnp,
 		if (free_slot)
 			__wt_log_slot_free(session, myslot.slot);
 	} else if (force) {
-		/*
-		 * If we are going to wait for this slot to get written,
-		 * signal the wrlsn thread.
-		 *
-		 * XXX I've seen times when conditions are NULL.
-		 */
-		if (conn->log_cond != NULL) {
-			__wt_cond_signal(session, conn->log_cond);
-			__wt_yield();
-		} else
-			WT_ERR(__wt_log_force_write(session, 1, NULL));
+		__wt_cond_signal(session, &conn->log_cond);
+		__wt_yield();
```

Comment:
The `log_cond` is used with the internal log server thread.  So it was allocated and initialized when we created the `log_server` thread in `wt_logmgr_open`.  It is in there (instead of `wt_logmgr_create`) because the user may get into this code via reconfigure.  So I do think we can come through here still without initialization because recovery/checkpoint writes log records before `logmgr_open`.  That is we've opened the underlying logging subsystem but haven't started all the worker threads.  Recovery and writing a log record does not need all the worker threads.  Or refactor to always initialize the condition var in `logmgr_create` but reconfigure needs consideration.